### ISSUE TITLE
Add summary data API for each DOI prefix + root level

### ISSUE THREAD

**Issue by blahah** [fathomlabs/open-retractions#7]
When we update the data, we should also calculate summary statistics for the subtree of each directory.

## api structure

This would provide a nice simple API for getting summary statistics:

- summary of all data: `http://openretractions.com/api/summary.json`
- summary of all DOI data: `http://openretractions.com/api/doi/summary.json`
- summary of all data for DOI prefix `10.7860`: `http://openretractions.com/api/doi/10.7860/summary.json`

## data structure

Every summary `summary.json` should include:

- `total`: the total number of unique retractions under this level
- `counts`: an object mapping each immediate child level to the number of retractions
- `publishers`: object with count of retractions by journal
- `journals`: object with count of retractions by journal
- `authors`: object with count of retractions by author ?

**Comment by egonw** [comment]
(Some statistics could be exposed on the human-oriented web page too...)

**Comment by blahah** [comment]
@egonw yeah we were thinking of keeping the main page nice and clean, with a small link to a stats dashboard.

we could also allow people to search for e.g. a doi prefix, and get back the stats for that prefix
